src/main/java/io/aiven/kafka/connect/gcs/GcsSinkConfig.java

```diff
@@ -315,25 +325,39 @@ static Map<String, String> handleDeprecatedYyyyUppercase(final Map<String, Strin
     private void validate() {
         final String credentialsPath = getString(GCS_CREDENTIALS_PATH_CONFIG);
         final Password credentialsJson = getPassword(GCS_CREDENTIALS_JSON_CONFIG);
-        if (credentialsPath != null && credentialsJson != null) {
-            final String msg = String.format("\"%s\" and \"%s\" are mutually exclusive options, but both are set.",
-                    GCS_CREDENTIALS_PATH_CONFIG, GCS_CREDENTIALS_JSON_CONFIG);
-            throw new ConfigException(msg);
+        final Boolean defaultCredentials = getBoolean(GCS_CREDENTIALS_DEFAULT_CONFIG);
+
+        final long nonNulls = Stream.of(defaultCredentials, credentialsJson, credentialsPath)
+                .filter(Objects::nonNull)
+                .count();
+
+        // only validate non nulls here, since all nulls means falling back to the default "no credential" behavour.
+        if (nonNulls > MAX_ALLOWED_CREDENTIAL_CONFIGS) {
+            throw new ConfigException(String.format("Only one of %s, %s, and %s can be non-null.",
+                    GCS_CREDENTIALS_DEFAULT_CONFIG, GCS_CREDENTIALS_JSON_CONFIG, GCS_CREDENTIALS_PATH_CONFIG));
         }
     }
 
     public OAuth2Credentials getCredentials() {
         final String credentialsPath = getString(GCS_CREDENTIALS_PATH_CONFIG);
         final Password credentialsJsonPwd = getPassword(GCS_CREDENTIALS_JSON_CONFIG);
-        if (credentialsPath == null && credentialsJsonPwd == null) {
+        final Boolean defaultCredentials = getBoolean(GCS_CREDENTIALS_DEFAULT_CONFIG);
+
+        // if we've got no path, json and not configured to use default credentials, fall back to connecting without
+        // any credentials at all.
+        if (credentialsPath == null && credentialsJsonPwd == null
+                && (defaultCredentials == null || !defaultCredentials)) {
             LOG.warn("No GCS credentials provided, trying to connect without credentials.");
             return NoCredentials.getInstance();
         }
+
         try {
             String credentialsJson = null;
             if (credentialsJsonPwd != null) {
                 credentialsJson = credentialsJsonPwd.value();
             }
+            // in the case where both path and json are empty the GoogleCredentialsBuilder will fall back to
+            // connecting with default credential discovery.
             return GoogleCredentialsBuilder.build(credentialsPath, credentialsJson);
```

Comment:
as this is not testeable, I'm not sure if it's better to do the validation here and use `GoogleCredentials.getApplicationDefault()` directly if both path and json are empty, wdyt?

Reply:
PR updated with this implementation.